reflectx/reflect.go

```diff
@@ -6,20 +6,63 @@
 //
 package reflectx
 
-import "sync"
-
 import (
+	"fmt"
 	"reflect"
 	"runtime"
+	"strings"
+	"sync"
 )
 
-type fieldMap map[string][]int
+type field struct {
+	Index    []int
+	Path     string
+	Field    reflect.StructField
+	Zero     reflect.Value
+	Name     string
+	Options  map[string]string
+	Embedded bool
+}
+
+type fields struct {
+	Index []*field
+	Paths map[string]*field
+	Names map[string]*field
+}
+
+func (f fields) GetByPath(path string) *field {
```

Comment:
thats true, they're more useful to help to build the fields.Paths map, but it could be useful to return the Path while traversing the index (tree). Ie. technically, reflectx could parse a json struct tag tree and returning the full path of a field may be useful there.